### PR TITLE
fix: horizontal overflow rtl scroll behavior and add rtl scrollLeft utility

### PR DESCRIPTION
**Pull request by scomea**
# Description
HorizontalOverflow has issues with scrolling in rtl because of inconsitencies in how browsers implement Element.scrollLeft:
http://lists.w3.org/Archives/Public/www-style/2014Apr/0438.html

The fix involved creating a RtlScrollConverter utility which provides function calls which standardize the behavior of getting/setting scroll left in rtl.  We are modelling the "go negative as you scroll in rtl" as the "correct" behavior based on https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollLeft

## Motivation & context
https://github.com/microsoft/fast-dna/issues/1559
https://github.com/microsoft/fast-dna/issues/1560
https://github.com/microsoft/fast-dna/issues/2126
https://github.com/microsoft/fast-dna/issues/2402
https://github.com/microsoft/fast-dna/issues/2437

## Issue type checklist
- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

**Comment by scomea** [comment]
Note: this PR adjusts test coverage thresholds slightly - branches from 96% to 95%, and lines from 98% to 97%.

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx [hunk 1]
```diff
@@ -274,19 +279,18 @@ class HorizontalOverflow extends Foundation<
...
+        if (this.getScrollPosition() === 0) {
+            return { start: false, end: true };
         } else if (
             distanceRemaining === this.horizontalOverflowItemsRef.current.clientWidth
         ) {
-            return { start: !isLtr, end: isLtr };
+            return { start: true, end: false };
         } else {
-            return { start: false, end: false };
+            return { start: true, end: true };
```
Comment (janechu) [inline packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: Nit: I would find this a little more readable.

```
const scrollPositionIsAtBeginning: boolean = this.getScrollPosition() === 0;
const scrollPositionIsAtEnd: boolean = distanceRemaining === this.horizontalOverflowItemsRef.current.clientWidth;
const scrollPositionIsInMiddle: boolean = !scrollPositionIsAtEnd && !scrollPositionIsAtBeginning;

return {
    start: scrollPositionIsInMiddle
        ? false
        : scrollPositionIsAtBeginning,
    end: scrollPositionIsInMiddle
        ? false
        : scrollPositionIsAtEnd
}
```

File: packages/fast-web-utilities/src/localization.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { canUseDOM } from "exenv-es6";
```
Comment (janechu) [inline packages/fast-web-utilities/src/localization.ts]: why include this import?

Comment (scomea) [reply packages/fast-web-utilities/src/localization.ts]: removed

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx (hunk 1 above)
Comment (scomea) [reply packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: yep

**Pull request merged**